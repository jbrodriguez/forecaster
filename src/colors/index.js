// @flow

import adapter from '../lib/adapter'

const palette = {
  primary: '#878D93', // baseText
  secondary: '#B9BCC0', // altText
  accent: '#ffffff', // accentText

  background: '#0F1B27',
  bgTone1: '#2F3B46',
  bgTone2: '#1A2733',

  contrast: '#6D6345',

  divider: '#1E2934',
}

const colors = adapter(palette)

export default colors
