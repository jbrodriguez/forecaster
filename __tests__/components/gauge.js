import { StyleSheet } from 'react-native'
import React from 'react'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

import Tachyons from 'react-native-style-tachyons'

import colors from '../../src/colors'
import Gauge from '../../src/components/gauge'

describe('Gauge', () => {
  Tachyons.build({ rem: 16 }, StyleSheet)

  const gui = { s: Tachyons.styles, c: colors, z: Tachyons.sizes }
  const measure = { value: 32, unit: 'metric' }

  test('renders correctly - nohero', () => {
    const tree = renderer.create(<Gauge gui={gui} measure={measure} caption="temperature" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly - hero', () => {
    const tree = renderer.create(<Gauge gui={gui} measure={measure} caption="temperature" hero />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
