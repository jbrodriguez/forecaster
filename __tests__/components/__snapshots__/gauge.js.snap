// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gauge renders correctly - hero 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 24,
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      32
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 16,
          },
          Object {
            "color": "rgba(255,255,255,0.7)",
          },
        ]
      }
    >
      metric
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 16,
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "color": "#CBC52A",
          },
        ]
      }
    >
      temperature
    </Text>
  </View>
</View>
`;

exports[`Gauge renders correctly - nohero 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 20,
          },
          Object {
            "color": "#B9BCC0",
          },
        ]
      }
    >
      32
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 16,
          },
          Object {
            "color": "#878D93",
          },
        ]
      }
    >
      metric
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 16,
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "color": "#2F3B46",
          },
        ]
      }
    >
      temperature
    </Text>
  </View>
</View>
`;
